pkg/view/router.go

```diff
@@ -54,17 +109,220 @@ func New(c config.DashboardConfig, auth func(h http.Handler) http.Handler) (*chi
+			return
+		}
+		destinations, err := storageServices.Database.GetDestinations(r.Context(), user.ID)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		m := loadModel(r)
+		m.Connections = Connections{
+			Destinations: destinations,
+		}
+		err = gv.Render(w, http.StatusOK, "pages/connections/index", m)
 		if err != nil {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 		}
 	})
 
 	r.Get("/connections/new", func(w http.ResponseWriter, r *http.Request) {
-		err := gv.Render(w, http.StatusOK, "pages/connections/new", loadModel(r))
+		err := gv.Render(w, http.StatusOK, "pages/connections/upsert", loadModel(r))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+	})
+
+	r.Post("/connections/upsert", func(w http.ResponseWriter, r *http.Request) {
+		user, ok := getUser(r)
+		if !ok {
+			http.Error(w, "User not found", http.StatusInternalServerError)
+			return
+		}
+
+		err := r.ParseForm()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		id := r.Form.Get("id")
+		if id != "" {
+			http.Error(w, "Editing connections not yet supported", http.StatusBadRequest)
+			return
+		}
+
+		settings := map[string]any{}
+		name := r.Form.Get("name")
+
+		t := r.Form.Get("type")
+		switch t {
+		case "duckdb":
+			// XXX breadchris support file destination?
+
+			tok := r.Form.Get("token")
+			if tok == "" {
+				http.Error(w, "Must specify a token", http.StatusBadRequest)
+				return
+			}
+			db := r.Form.Get("database")
+			if db == "" {
+				http.Error(w, "Must specify a token", http.StatusBadRequest)
+				return
+			}
+
+			// XXX breadchris what validation should be done here?
+
+			settings["token"] = tok
+			settings["database"] = db
+		default:
+			http.Error(w, "Unknown connection type", http.StatusBadRequest)
+			return
+		}
+
+		cd := config.Destination{
+			Type:     t,
+			Settings: settings,
+		}
+
+		err = destManager.TestCredentials(cd)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		teamId, err := storageServices.Database.GetTeamId(user.ID)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		dest, err := storageServices.Database.CreateDestination(r.Context(), teamId, name, t, settings)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		key := uuid.New().String()
+		hashedKey := storageServices.Database.Hash(key)
+		err = storageServices.Database.AddAPIKey(r.Context(), dest.ID, hashedKey)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		m := loadModel(r)
+		m.Connect.InsertExample = fmt.Sprintf(
+			"curl -X POST '%s/api/data/insert/%s?api_key=%s' --json '{\"user\": \"bob\", \"event\": \"click\"}'",
+			c.ExternalURL,
+			dest.Name,
```

Comment:
This is confusing. What if the name has spaces? I'd just use an obvious value like `your_table`.

Reply:
if the name of the table is `your_table` then the copy and paste wouldn't work until they edit in their own table name. Is that what you had in mind?

Reply:
I'd say we give a default value, like `scratchdata_test` or `scratchdata` or even `your_table`. The software will automatically create the table if it doesn't exist when they do the insert, and perhaps that will help users see the end-to-end flow.

Reply:
ah ok, I think I had a misunderstanding of what a table was. This makes sense.